src/diff/children.js

```diff
@@ -40,27 +46,148 @@ export function diffChildren(
 	refQueue
 ) {
 	let i,
-		j,
 		/** @type {VNode} */
 		oldVNode,
 		/** @type {VNode} */
 		childVNode,
 		/** @type {PreactElement} */
 		newDom,
 		/** @type {PreactElement} */
-		firstChildDom,
-		skew = 0;
+		firstChildDom;
 
 	// This is a compression of oldParentVNode!=null && oldParentVNode != EMPTY_OBJ && oldParentVNode._children || EMPTY_ARR
 	// as EMPTY_OBJ._children should be `undefined`.
 	/** @type {VNode[]} */
 	let oldChildren = (oldParentVNode && oldParentVNode._children) || EMPTY_ARR;
 
-	let oldChildrenLength = oldChildren.length,
-		remainingOldChildren = oldChildrenLength,
-		newChildrenLength = renderResult.length;
+	let newChildrenLength = renderResult.length;
+
+	// TODO: Could we drop oldDom all together and just use _nextDom?
```

Comment:
We could check this in a follow-up PR

Reply:
Agreed. I gave it a quick attempt and it got messy really quickly. So not gonna do that in this PR. Will clean up the comment before checking in